From fa4b74f05dacb3c4a2a1dda6320e1b6c7ea600f0 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Sun, 7 Nov 2021 08:33:36 -0500
Subject: [PATCH 16/21] Disable aligned-allocation on macOS

include/c++/v1/memory:1423:5:
  error: aligned deallocation function of type
  'void (void *, std::align_val_t) noexcept'
  is only available on macOS 10.14 or newer
    delete __ptr;
    ^
include/c++/v1/memory:1684:7:
  note: in instantiation of member function
  'std::__1::default_delete<mongo::ReshardingMetrics::OperationMetrics>::operator()'
  requested here
      __ptr_.second()(__tmp);
      ^
include/c++/v1/memory:1642:5:
  note: in instantiation of member function
  'std::__1::unique_ptr<mongo::ReshardingMetrics::OperationMetrics,
  std::__1::default_delete<mongo::ReshardingMetrics::OperationMetrics>>::reset'
  requested here
    reset();
    ^
src/mongo/db/s/resharding/resharding_metrics.cpp:400:16:
  note: in instantiation of member function
  'std::__1::unique_ptr<mongo::ReshardingMetrics::OperationMetrics,
  std::__1::default_delete<mongo::ReshardingMetrics::OperationMetrics>>::operator='
  requested here
    _currentOp = nullptr;
               ^

Drop this patch when MACOSX_DEPLOYMENT_TARGET is updated to 10.14+
---
 SConstruct | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SConstruct b/SConstruct
index fccf7db53b3..01609727ccd 100644
--- a/SConstruct
+++ b/SConstruct
@@ -3153,6 +3153,10 @@ def doConfigure(myenv):
         if not env.TargetOSIs("macOS"):
             myenv.AddToCXXFLAGSIfSupported("-fsized-deallocation")
 
+        # Requires macOS >=10.14
+        if env.TargetOSIs("macOS"):
+            myenv.AddToCXXFLAGSIfSupported("-fno-aligned-allocation")
+
         # This warning was added in Apple clang version 11 and flags many explicitly defaulted move
         # constructors and assignment operators for being implicitly deleted, which is not useful.
         myenv.AddToCXXFLAGSIfSupported("-Wno-defaulted-function-deleted")
